src/server/req_preemptjob.c

```diff
@@ -385,10 +386,23 @@ reply_preempt_jobs_request(int code, int aux, struct job *pjob)
 		if (pjob->preempt_order[0].order[pjob->preempt_order_index] != PREEMPT_METHOD_LOW) {
 			if (issue_preempt_request((int)pjob->preempt_order[0].order[pjob->preempt_order_index], pjob, preq)) {
 				job_preempt_fail(preq, pjob->ji_qs.ji_jobid);
+				pjob->preempt_order_index = 0;
+				pjob->preempt_order = NULL;
 				pjob->ji_pmt_preq = NULL;
+			} else {
+				/* reply_preempt_jobs_request() is somewhat recursive.  It is possible for one call to issue the next
```

Comment:
pjob->ji_pmt_preq will only be NULL if we reached the end of the preemption methods and the preemption failed. In the case where the next preemption method was not immediately rejected, we won't return.

Reply:
Right, but the else{} block will be executed when issue_preempt_request() is successful right? Will issue_preempt_request() return success if pjob->ji_pmt_preq is NULL ?

Reply:
issue_Drequest() can only fail in the function itself.  If the dependent call (e.g. req_signaljob()) rejects the request, it still returns success.  The only way we know all preemption methods have failed is because ji_pmt_preq is NULL.  Let me see if I can reword the comment some (it's already 6 lines long) to explain this better.